inference/conflict.go

```diff
@@ -69,16 +69,16 @@ func newNode(p annotation.Prestring, c annotation.Prestring) node {
 
 func (n *node) String() string {
 	posStr := "<no pos info>"
-	reasonStr := "<no reason info>"
+	reasonStr := ""
 	if n.consumerPosition.IsValid() {
 		posStr = n.consumerPosition.String()
 	}
-	if len(n.producerRepr) > 0 && len(n.consumerRepr) > 0 {
-		reasonStr = n.producerRepr + " " + n.consumerRepr
-	} else if len(n.producerRepr) > 0 {
-		reasonStr = n.producerRepr
-	} else if len(n.consumerRepr) > 0 {
-		reasonStr = n.consumerRepr
+
+	if len(n.producerRepr) > 0 {
+		reasonStr += n.producerRepr + " "
+	}
+	if len(n.consumerRepr) > 0 {
+		reasonStr += n.consumerRepr
```

Comment:
So this means if `n.producerRepr` is non-empty and `n.consumerRepr` is empty, we would result in a string `<producerRepr> ` (notice the trailing space).

Not that it matters much from UX perspective, but perhaps we do not want that anyways?

Reply:
Good catch. Fixed.